Distribution/PackDeps.hs

```diff
@@ -117,22 +111,36 @@ addPackage m entry =
     go package' version =
         case Tar.entryContent entry of
             Tar.NormalFile bs _ ->
-                 Map.insert package' PackInfo
+                Map.insertWith maxVersion package' PackInfo
                         { piVersion = version
                         , piDesc = parsePackage bs
                         , piEpoch = Tar.entryTime entry
                         } m
             _ -> m
 
 data PackInfo = PackInfo
-    { piVersion :: Version
-    , piDesc :: Maybe DescInfo
-    , piEpoch :: Tar.EpochTime
+    { piVersion  :: Version
+    , piDesc     :: Maybe DescInfo
+    , piEpoch    :: Tar.EpochTime
     }
     deriving (Show, Read)
 
+-- Extract revision from PackInfo, default to 0
+piRevision :: PackInfo -> Int
+piRevision = fromMaybe 0 . fmap diRevision . piDesc
+
 maxVersion :: PackInfo -> PackInfo -> PackInfo
-maxVersion pi1 pi2 = if piVersion pi1 <= piVersion pi2 then pi2 else pi1
```

Comment:
Given that we're _not_ actually doing the comparison on `piRevision`, is there any purpose to this change?